packages/app-mobile/components/NoteEditor/ImageEditor/ImageEditor.tsx

```diff
@@ -0,0 +1,205 @@
+						await clearAutosave();
+						props.onCancel();
+					},
+					style: 'destructive',
+				},
+				{
+					text: _('Save changes'),
+					onPress: () => {
+						// saveDrawing calls props.onSave(...) which may close the
+						// editor.
+						webviewRef.current.injectJS('saveDrawing();');
+					},
+				},
+			]
+		);
+		return true;
+	}, [webviewRef, props.onCancel]);
+
+	useEffect(() => {
+		BackHandler.addEventListener('hardwareBackPress', onRequestCloseEditor);
+
+		return () => {
+			BackHandler.removeEventListener('hardwareBackPress', onRequestCloseEditor);
+		};
+	}, [onRequestCloseEditor]);
+
+	const css = useCss(editorTheme);
+	const html = useMemo(() => `
+		<!DOCTYPE html>
+		<html>
+			<head>
+				<meta charset="utf-8"/>
+				<meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1,user-scalable=no"/>
+
+				<style>
+					${css}
+				</style>
+			</head>
+			<body></body>
+		</html>
+	`, [css]);
+
+	const injectedJavaScript = useMemo(() => `
+		window.onerror = (message, source, lineno) => {
+			window.ReactNativeWebView.postMessage(
+				"error: " + message + " in file://" + source + ", line " + lineno
+			);
+		};
+
+		const saveDrawing = (isAutosave) => {
+			const img = window.editor.toSVG();
+			window.ReactNativeWebView.postMessage(
+				JSON.stringify({
+					action: isAutosave ? 'autosave' : 'save',
+					data: img.outerHTML,
+				}),
+			);
+		};
+		window.saveDrawing = saveDrawing;
+
+		const closeDrawing = () => {
+			window.ReactNativeWebView.postMessage('{ "action": "close" }');
+		};
+
+		try {
+			if (window.editor === undefined) {
+				${shim.injectedJs('svgEditorBundle')}
+
+				window.editor = svgEditorBundle.createJsDrawEditor();
+
+				window.initialSVGData = ${JSON.stringify(props.initialSVGData)};
+				if (initialSVGData && initialSVGData.length > 0) {
+					editor.loadFromSVG(initialSVGData);
+				}
+
+				const toolbar = editor.addToolbar();
+
+				toolbar.addActionButton({
+					label: ${JSON.stringify(_('Close'))}, 
+					icon: svgEditorBundle.makeCloseIcon(),
+				}, () => closeDrawing());
+
+				toolbar.addActionButton({
+					label: ${JSON.stringify(_('Done'))},
+					icon: editor.icons.makeSaveIcon(),
+				}, () => {
+					saveDrawing(false);
+				});
+
+				svgEditorBundle.restoreToolbarState(
+					toolbar,
+					${JSON.stringify(Setting.value('imageeditor.jsdrawToolbar'))}
+				);
+				svgEditorBundle.listenToolbarState(editor, toolbar);
+
+				// Auto-save every two minutes.
+				const autoSaveInterval = 2 * 60 * 1000;
+				setInterval(() => {
+					saveDrawing(true);
+				}, autoSaveInterval);
+			}
+		} catch(e) {
+			window.ReactNativeWebView.postMessage(
+				'error: ' + e.message + ': ' + JSON.stringify(e)
+			);
+		}
+		true;
+	`, [props.initialSVGData]);
+
+	const onMessage = useCallback(async (event: WebViewMessageEvent) => {
+		const data = event.nativeEvent.data;
+		if (data.startsWith('error:')) {
+			console.error('ImageEditor:', data);
```

Comment:
Could you use logger.error() for log messages please? You can create a logger by adding `const logger = Logger.create('ImageEditor')` at the top of the file